orbit/pkg/profiles/profiles_darwin_test.go

```diff
@@ -24,14 +24,31 @@ func TestGetFleetdConfig(t *testing.T) {
 		{ptr.String("invalid-json"), nil, nil, "unmarshaling configuration"},
 		{ptr.String("{}"), nil, nil, ErrNotFound.Error()},
 		{
-			ptr.String(`{"EnrollSecret": "ENROLL_SECRET", "FleetURL": "https://test.example.com"}`),
+			ptr.String(`{"EnrollSecret": "ENROLL_SECRET", "FleetURL": "https://test.example.com", "EnableScripts": true}`),
 			nil,
 			&fleet.MDMAppleFleetdConfig{
-				EnrollSecret: "ENROLL_SECRET",
-				FleetURL:     "https://test.example.com",
+				EnrollSecret:  "ENROLL_SECRET",
+				FleetURL:      "https://test.example.com",
+				EnableScripts: true,
 			},
 			"",
 		},
+		{
+			ptr.String(`{"EnrollSecret": "ENROLL_SECRET", "FleetURL": "https://test.example.com", "EnableScripts": false}`),
+			nil,
+			&fleet.MDMAppleFleetdConfig{
+				EnrollSecret:  "ENROLL_SECRET",
+				FleetURL:      "https://test.example.com",
+				EnableScripts: false,
+			},
+			"",
+		},
+		{
+			ptr.String(`{"EnableScripts": true}`),
+			nil,
+			nil,
+			ErrNotFound.Error(),
+		},
```

Comment:
this is an important implication: we consider a configuration profile of type `com.fleetdm.fleetd.config` to be valid only if it contains an enroll secret and a fleet url.

more concretely, this means that you can't enable script execution via config profile unless you also set the other two values.

this makes things easier for us, but I'm not sure if it's acceptable. Any thoughts? cc: @noahtalerman

Reply:
> you can't enable script execution via config profile unless you also set the other two values

@roperzh got it. 

How does this affect our customer that's deploying a config profile with enroll secret and Fleet URL on their own? How will they enable script execution?

Also, will script execution be enabled if you set any enroll secret and fleet URL in `com.fleetdm.fleetd.config`? Or do they have to be valid? (aka the host has to be able to connect to a Fleet server)

Reply:
> how does this affect our customer that's going to deploy a config profile with enroll secret and Fleet URL on their own? How will they enable script execution?

this would only affect the potential scenario of a customer trying to enable script execution via a configuration profile without setting the other two values.

since this customer is planning to set both fleet url and enroll secret, they only have to add a third key to their `mobileconfig` with:

```xml
<key>ScriptsEnabled</key>
<true />
```

> Also, will script execution be enabled if you set any enroll secret and fleet URL in com.fleetdm.fleetd.config? Or do they have to be valid? (aka the host has to be able to connect to a Fleet server)

the answer is two folded:

1. It will be enabled in the sense that the binary will be "ready" to execute scripts
2. It won't be enabled since `fleetd` receives the scripts to execute from the server, and it won't be able to retrieve any scripts if it doesn't have a valid URL and enroll secret

I hope that makes sense!